api/src/test/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImplTest.java

```diff
@@ -0,0 +1,133 @@
+package org.pmiops.workbench.cohortbuilder;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+import javax.inject.Provider;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.pmiops.workbench.FakeClockConfiguration;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.cdr.cache.MySQLStopWords;
+import org.pmiops.workbench.cdr.dao.CBCriteriaAttributeDao;
+import org.pmiops.workbench.cdr.dao.CBCriteriaDao;
+import org.pmiops.workbench.cdr.dao.CBDataFilterDao;
+import org.pmiops.workbench.cdr.dao.CriteriaMenuDao;
+import org.pmiops.workbench.cdr.dao.DomainCardDao;
+import org.pmiops.workbench.cdr.dao.PersonDao;
+import org.pmiops.workbench.cdr.dao.SurveyModuleDao;
+import org.pmiops.workbench.cohortbuilder.mapper.CohortBuilderMapper;
+import org.pmiops.workbench.cohortbuilder.mapper.CohortBuilderMapperImpl;
+import org.pmiops.workbench.test.FakeClock;
+import org.pmiops.workbench.workspaces.WorkspaceAuthService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.annotation.DirtiesContext;
+
+@DataJpaTest
```

Comment:
Does it makes sense to make this a DataJpaTest when it's not using the in memory database? Seems like alot of overhead for nothing.

Reply:
Initially I did it all as mocks, since we are only testing a util method. However I opted to do the heavy weight since the test class is CohortBuilderServiceImplTest, so in future, if tests need to be written all daos are already wired.  I can change all to be mocks and test a non-jpa test.  Strictly, these methods, including other validate* methods in other service/controller classes should be refactored into utilitiy class(es) with static methods. But your call.. Let me know

Reply:
I think this is fine. thanks